setup.py

```diff
@@ -388,7 +388,10 @@ def run_setup(extensions):
         # 1.) build_ext eats errors at compile time, letting the install complete while producing useful feedback
         # 2.) there could be a case where the python environment has cython installed but the system doesn't have build tools
         if pre_build_check():
-            kw['setup_requires'] = ['Cython>=0.20,<0.25']
+            user_specified_cython_version = os.environ.get('CASS_DRIVER_ALLOWED_CYTHON_VERSION')
+            kw['setup_requires'] = ['Cython>=0.20,<0.25' if
+                                    user_specified_cython_version is None else
+                                    'Cython=={}'.format(user_specified_cython_version)]
```

Comment:
for clarity if we need to add more deps in the future. I would do something like: 
```python
cython_dep = 'Cython>=0.20,<0.25'
user_specified_cython_version = os.environ.get('CASS_DRIVER_ALLOWED_CYTHON_VERSION')
if user_specified_cython_version:
    cython_dep =  'Cython==%s' % user_specified_cython_version

kw['setup_requires'] = [cython_dep]
```

Note that In the whole code base, we use the old style formatting ('%s' % 'hello'), so better to stick to that here as well.